rpc_util.go

```diff
@@ -719,15 +719,19 @@ func outPayload(client bool, msg any, data, payload []byte, t time.Time) *stats.
 	}
 }
 
-func checkRecvPayload(pf payloadFormat, recvCompress string, haveCompressor bool) *status.Status {
+func checkRecvPayload(pf payloadFormat, recvCompress string, haveCompressor bool, isServer bool) *status.Status {
 	switch pf {
 	case compressionNone:
 	case compressionMade:
 		if recvCompress == "" || recvCompress == encoding.Identity {
 			return status.New(codes.Internal, "grpc: compressed flag set with identity or empty encoding")
 		}
 		if !haveCompressor {
-			return status.Newf(codes.Unimplemented, "grpc: Decompressor is not installed for grpc-encoding %q", recvCompress)
+			if isServer {
+				return status.Newf(codes.Unimplemented, "grpc: Decompressor is not installed for grpc-encoding %q", recvCompress)
+			} else {
+				return status.Newf(codes.Internal, "grpc: Decompressor is not installed for grpc-encoding %q", recvCompress)
```

Comment:
I think I'm fine with punting on this "requirement" entirely.

The client logging what the client supports doesn't seem critical.  Let's just fix the status code in this PR?

I'm mostly just worried about option 3 misleading people, since it would be incomplete (i.e. not including the `ForceCompressor`'s (/etc) name).

Probably the "best" fix would be to have a per-channel codec registry that is initialized to the global and augmented with the dial options (and further augmented with the call options for tracking per-call).  But that's too much work to be worth it IMO.

Reply:
Ack.